### PR TITLE
r_cons_print fix for very large output ##cons

### PR DESCRIPTION
**Pull request by HoundThe**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md#code-style)

**Detailed description**

There was a segfault linked to a very large binary with DWARF information (http://xvilka.me/vmlinux_ida_dwarf_fail.zip) when using `id`. It was caused by the information being over several GiB of text and r_cons_context using `int` for it's size and length, changing it to ut64 fixed the segfaulting behaviour.

### CODE REVIEW

File: libr/include/r_cons.h [hunk 1]
```diff
@@ -447,8 +447,8 @@ typedef struct r_cons_context_t {
 	RConsGrep grep;
 	RStack *cons_stack;
 	char *buffer;
```
Comment (trufae) [inline libr/include/r_cons.h:449]: This should be done using RBuffer ideally.

File: libr/include/r_cons.h [hunk 2]
```diff
@@ -447,8 +447,8 @@ typedef struct r_cons_context_t {
...
-	int buffer_len;
-	int buffer_sz;
+	ut64 buffer_len;
```
Comment (trufae) [inline libr/include/r_cons.h:450]: What about size_t?

Comment (HoundThe) [reply libr/include/r_cons.h:450]: Sure, could be size_t aswell probably, I think size_t is guaranteed to hold size of any object so it should cover capacity of whole possible memory.

**Comment by HoundThe** [comment]
> Which situation the buffer is thst big ?

As mentioned in the PR, for a large binary with a lot of DWARF debugging information, this happens when printing all the information out.

File: libr/include/r_cons.h (hunk 2 above)
Comment (ret2libc) [reply libr/include/r_cons.h:450]: however size_t could be 32bit on 32bit system. Would this risk to reintroduce the wrong behaviour on those platforms?

Comment (HoundThe) [reply libr/include/r_cons.h:450]: well 32bit systems wouldn't fit more 2^32 bytes of output anyways

Comment (ret2libc) [reply libr/include/r_cons.h:450]: Good point :D :D

Comment (trufae) [reply libr/include/r_cons.h:450]: use size_t and u get my +1

**Pull request merged**